compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -7539,6 +7539,7 @@ private BLangClassDefinition desugarTemplateLiteralObjectTypedef(List<BLangLiter
         // Create a new concrete, class type for the provided abstract object type
         BObjectType objectClassType = new BObjectType(classTSymbol, classTSymbol.flags);
         objectClassType.fields = objectType.fields;
+        objectClassType.originalFields = objectType.fields;
```

Comment:
This kind of assignments are problematic since you're setting the same value to both fields. Changes done using `fields` will be visible through `originalFields` as well. Applies to all the other places as well.